internal/pkg/telemetry/ddtracer.go

```diff
@@ -15,6 +20,13 @@ type wrappedDDTracer struct {
 func (t *wrappedDDTracer) Start(parentCtx context.Context, spanName string, opts ...trace.SpanStartOption) (ctx context.Context, span trace.Span) {
 	ctx, span = t.Tracer.Start(parentCtx, spanName, opts...)
 	if span != nil {
+		spanCtx := span.(ddtrace.Span).Context()
```

Comment:
`span.(ddtrace.Span)` panics if the type doesn't match interface.

Telemetry is an optional part of the syste, safer variant is:
```
if spanCtx, ok := span.(ddtrace.Span).Context(); ok {
  ...
}
```

Reply:
yes, sorry